lib/Command/Prompt.php

```diff
@@ -43,16 +45,55 @@ protected function configure() {
 				InputArgument::REQUIRED,
 				'The prompt'
 			)
-			->addOption('no-context', null, InputOption::VALUE_NONE, 'Do not use context');
+			->addOption(
+				'no-context',
+				null,
+				InputOption::VALUE_NONE,
+				'Do not use context'
+			)
+			->addOption(
+				'context-sources',
+				null,
+				InputOption::VALUE_REQUIRED,
+				'Context sources to use',
+			)
+			->addOption(
+				'context-providers',
+				null,
+				InputOption::VALUE_REQUIRED,
+				'Context provider to use',
+			);
 	}
 
 	protected function execute(InputInterface $input, OutputInterface $output) {
 		$userId = $input->getArgument('uid');
 		$prompt = $input->getArgument('prompt');
 		$noContext = $input->getOption('no-context');
+		$contextSources = $input->getOption('context-sources');
+		$contextProviders = $input->getOption('context-providers');
+
+		if ($noContext && (!empty($contextSources) || !empty($contextProviders))) {
+			throw new \InvalidArgumentException('Cannot use --no-context with --context-sources or --context-provider');
+		}
+
+		if (!empty($contextSources) && !empty($contextProviders)) {
```

Comment:
Be careful with `empty`: 



    The following things are considered to be empty:

        "" (an empty string)
        0 (0 as an integer)
        0.0 (0 as a float)
        "0" (0 as a string)
        NULL
        FALSE
        array() (an empty array)
        var $var; (a variable declared, but without a value in a class)



https://stackoverflow.com/a/2220529

Reply:
we're expecting only strings (providers or sources), so it serves the purpose well I think. Nice gotcha to lookout for though!